test/e2e/framework/statefulset_utils.go

```diff
@@ -355,7 +355,8 @@ func (s *StatefulSetTester) SetHealthy(ss *apps.StatefulSet) {
 	}
 }
 
-func (s *StatefulSetTester) waitForStatus(ss *apps.StatefulSet, expectedReplicas int32) {
+// WaitForStatus waits for the ss.Status.Replicas to be equal to expectedReplicas
+func (s *StatefulSetTester) WaitForStatus(ss *apps.StatefulSet, expectedReplicas int32) {
```

Comment:
@kargakis Using this code 
```golang
if ss.Status.ObservedGeneration == nil { 
        Logf("ss observed generation is nil")
} else {
        Logf("ss observed generation %d",ss.Status.ObservedGeneration)
}
Logf("ss generation %d",ss.Generation)
if ssGet.Status.ObservedGeneration == nil {
        Logf("ssGet observed generation is nil")
} else {
        Logf("ssGet observed generation %d",ssGet.Status.ObservedGeneration)
}
Logf("ssGet generation %d",ssGet.Generation)
```
The observed generation of both ss and ssGet are always nil. However the generation appears to increment consistently.

what about the following
```golang
if  ssGet.Generation < ss.Generation {
     return false, nil
}
```